src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -89,22 +89,17 @@ internal sealed class VariableInformation : IVariableInformation
         private string DisplayHint { get; set; }
         public bool IsPreformatted { get; set; }
 
-        private static readonly string[] s_validAddressFormats = new string[] {
-            @"^0x[0-9a-fA-F]+$",
-            @"^(0x[0-9a-fA-F]+)\b"
-        };
+        static readonly Lazy<Regex> s_addressPattern = new Lazy<Regex>(() => new Regex(@"^(0x[0-9a-fA-F]+)\b"));
```

Comment:
```suggestion
        static readonly Lazy<Regex> s_addressPattern = new Lazy<Regex>(() => new Regex(@"^(0x[0-9a-fA-F]+)\b", RegexOptions.Compiled));
```
Compile Regex for performance. https://docs.microsoft.com/en-us/dotnet/standard/base-types/compilation-and-reuse-in-regular-expressions

Reply:
This feels like it is too rarely used to be worth using.

---
In reply to: [594648798](https://github.com/microsoft/MIEngine/pull/1117#discussion_r594648798) [](ancestors = 594648798)